### ISSUE TITLE
Cabal new-build does not contain executable in dist-newstyle

### ISSUE THREAD

**Issue by Rizary** [haskell/cabal#4434]
So I have project folder in this https://github.com/Rizary/cabal-new-try repo.

it basically contains two folders (frontend and backend). I try to build the frontend folder which has its own cabal.project and use ghcjs compiler. 

cabal.project

```
packages: ./*.cabal
optional-packages: ./*/{reflex,reflex-dom,reflex-dom-contrib}/*.cabal

with-compiler: /Users/Rizitecs/.cabal/bin/ghcjs
compiler: ghcjs

```

I have deps folder which has reflex,reflex-dom,reflex-dom-contrib submodules.

when I do `cabal new-build` the build is fine, and the output is as follow:

```
Warning: don't know how to find change monitoring files for the installed
package databases for ghcjs
Resolving dependencies...
Warning: don't know how to find change monitoring files for the installed
package databases for ghcjs
In order, the following will be built (use -v for more details):
dependent-sum-0.3.2.2
dependent-map-0.2.4.0
dependent-sum-template-0.0.0.5
prim-uniq-0.1.0.1
reflex-0.5.0
reflex-dom-0.4
frontend-0.0.0.1
Configuring dependent-sum-0.3.2.2...
Building dependent-sum-0.3.2.2...
```

it does create folder dist-newstyle, but when I execute tree on the folder, it only shows:

```
dist-newstyle/
├── build
├── cache
│   ├── compiler
│   ├── config
│   ├── elaborated-plan
│   ├── improved-plan
│   ├── plan.json
│   ├── solver-plan
│   └── source-hashes
└── tmp
```

I have another project same as this, and it usually creates index.html and the .jsexe folder inside the dist/ folder.

**Comment by jparklev** [comment]
Any word on this issue?

**Comment by Rizary** [comment]
hi @jparklev I kinda forget about this (i am not good at remembering something). But it solved when  using   cabal 2.2 it works perfectly.

**Comment by hvr** [comment]
Fwiw, here's a real-world example from @nomeata's  https://github.com/nomeata/ghcjs2gh-pagesproving that things do work:

https://github.com/nomeata/ghcjs2gh-pages/blob/0ffa57424d45c6b2d406859844e433cd1525d203/.travis.yml#L61
